ipaserver/install/dogtaginstance.py

```diff
@@ -275,6 +279,64 @@ def uninstall(self):
             logger.critical("failed to uninstall %s instance %s",
                             self.subsystem, e)
 
+    def secure_ajp_connector(self):
+        """ Update AJP connector to use a password protection  """
+        server_xml = open(paths.PKI_TOMCAT_SERVER_XML, 'r').read()
+
+        # AJP connector is set on port 8009. Use non-greedy search to find it
+        connector = re.search('<Connector port="8009".+?/>', server_xml)
+
+        # no AJP connector means no need to update anything
+        if connector is None:
+            return
```

Comment:
Should this be an error condition? Isn't one mandatory or is this handled elsewhere?

Reply:
If we decide to remove AJP connector in future PKI releases, there is no need to secure it anymore, so this is an intentional logic here.